server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/CICSVisitor.java

```diff
@@ -89,6 +89,68 @@ public List<Node> visitQualifiedDataName(CICSParser.QualifiedDataNameContext ctx
         return addTreeNode(ctx, QualifiedReferenceNode::new);
     }
 
+    @Override
+    public List<Node> visitCics_handle_abend(CICSParser.Cics_handle_abendContext ctx) {
+        boolean isProgram =
+            Optional.ofNullable(ctx)
+                .map(CICSParser.Cics_handle_abendContext::PROGRAM)
+                .filter(s -> s.size() > 0)
+                .isPresent();
+
+        boolean isLabel =
+            Optional.ofNullable(ctx)
+                .map(CICSParser.Cics_handle_abendContext::LABEL)
+                .filter(s -> s.size() > 0)
+                .isPresent();
+
+        boolean isReset =
+            Optional.ofNullable(ctx)
+                .map(CICSParser.Cics_handle_abendContext::RESET)
+                .filter(s -> s.size() > 0)
+                .isPresent();
+
+        ExecCicsHandleNode.HandleAbendType type;
+        if (isProgram) {
+            type = ExecCicsHandleNode.HandleAbendType.PROGRAM;
+        } else if (isLabel) {
+            type = ExecCicsHandleNode.HandleAbendType.LABEL;
+        } else if (isReset) {
+            type = ExecCicsHandleNode.HandleAbendType.RESET;
+        } else {
+            type = ExecCicsHandleNode.HandleAbendType.CANCEL;
+        }
+
+        Node node = new ExecCicsHandleNode(getOriginalLocality(ctx.getParent().getParent().getParent()), type);
+        visitChildren(ctx).forEach(node::addChild);
+        return ImmutableList.of(node);
+    }
+
+    @Override
+    public List<Node> visitCics_abend(CICSParser.Cics_abendContext ctx) {
+        boolean isCancel = Optional.ofNullable(ctx.cics_abend_cancel()).map(l -> l.size()).orElse(0) > 0;
+        boolean isNodump = Optional.ofNullable(ctx.cics_abend_nodump()).map(l -> l.size()).orElse(0) > 0;
+        String[] abcodeArray = Optional.ofNullable(ctx.cics_abend_abcode())
+            .filter(l -> l.size() > 0)
+            .map(l -> l.get(0))
+            .map(a -> a.cics_name())
+            .map(a -> a.name())
+            .map(a -> a.variableNameUsage())
```

Comment:
This can also be just a variable - I am not sure how useful it is to capture the details.

Reply:
Fixed